odoo/modules/graph.py

```diff
@@ -178,7 +178,19 @@ def _pprint(self, depth=0):
         return s
 
     def should_have_demo(self):
-        return (hasattr(self, 'demo') or (self.dbdemo and self.state != 'installed')) and all(p.dbdemo for p in self.parents)
+        demo_enabled_addons = tools.config.get("demo_enabled_addons", "").split(",")
+        if demo_enabled_addons:
+            # Allow us to enable demo data for specific addons.
+            # For this to be called, ~/.odoorc will need e.g:
+            #
+            # [options]
+            # demo_enabled_addons = stock,sale
+            #
+            # This configuration option should _only_ ever be set in local dev or UAT instances.
+            return self.name in tools.config.get("demo_enabled_addons", "").split(",")
```

Comment:
`return self.name in demo_enabled_addons`

Reply:
We need the split since some module names are substrings of other modules.

Reply:
Cant this method inherited somewhere in our addons, like udes_common example?

Reply:
> We need the split since some module names are substrings of other modules.

`demo_enabled_addons` is already the split list; or am I missing something obvious...

Reply:
`demo_enabled_addons` is a string of addon names, comma separated e.g `demo_enabled_addons = custom_addon1,custom_addon2`

Reply:
> Cant this method inherited or overriden better somewhere in our addons, like udes_common example?

I don't think it can, it's a pure python class

Reply:
Rajeesh means that we have split already at line 181 and stored the result in  `demo_enabled_addons`, therefore we don't need to read the config again and split it, instead we can use his suggested code.

Reply:
oops, forgot i did that :laughing: sure i'll change tomorrow

Reply:
fixed: 9a4e29d009defde20eaa7576eeda41df958c11db